src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ComVariant.cs

```diff
@@ -558,7 +558,7 @@ public static unsafe ComVariant CreateRaw<T>(VarEnum vt, T rawValue)
         public VarEnum VarType
         {
             readonly get => (VarEnum)_typeUnion._vt;
-            private set => _typeUnion._vt = (ushort)value;
+            internal set => _typeUnion._vt = (ushort)value;
```

Comment:
Hey can we keep this `private` and instead construct a new instance of the `ComVariant` struct? Part of the design of this type was that you couldn't change the `VarType` out from under an existing instance.

Reply:
The current `ComVariant` constructors aren't friendly for this. Anyway I'm moving the initialization to native side.